### PR TITLE
Experiment one: basic.

### PR DESCRIPTION
**Pull request by MadcowD**
Cleaning master.

### CODE REVIEW

File: experiments/exp1/polynomial_critic.py [hunk 1]
```diff
@@ -86,44 +79,52 @@ def setup_graph(self, state_input, action_input, net, linear):
         print("state_input {0}. action_input {1}".format(state_input, action_input))
         return q_value_output, net
 
+    def create_poly_q(self):
+        """ Initialize the polynomial critic network"""
+        # TODO: FIX TO REFLECT NEW CHANGES TO OBJECT.
```
Comment (philkuz) [inline experiments/exp1/polynomial_critic.py:84]: Is this todo done?

File: experiments/exp1/polynomial_critic.py [hunk 2]
```diff
@@ -86,44 +79,52 @@ def setup_graph(self, state_input, action_input, net, linear):
         print("state_input {0}. action_input {1}".format(state_input, action_input))
         return q_value_output, net
 
+    def create_poly_q(self):
+        """ Initialize the polynomial critic network"""
+        # TODO: FIX TO REFLECT NEW CHANGES TO OBJECT.
+        # Create the variables.
+        linear = self.order <= 1
+        layer_size = self.state_dim + self.action_dim if not linear else 1
+        # TODO: ensure no conflict between dimension objects and ints
```
Comment (philkuz) [inline experiments/exp1/polynomial_critic.py:88]: Same here

File: experiments/exp1/polynomial_critic.py [hunk 3]
```diff
@@ -86,44 +79,52 @@ def setup_graph(self, state_input, action_input, net, linear):
...
+        q_value_output, net = self.setup_graph(self.state_placeholder , self.action_placeholder, net, linear)
 
+        #   then let x =tf.concat(state_placeholder, action_placeholder) and the output of this polynomial
+        #   critic will be Qn = x^TWx if order=2, or Qn = xW, if order =1, etc...
+        if self.summaries: variable_summaries(q_value_output, q_value_output.name)
+        return q_value_output, net
 
+    def create_target_q_network(self, net, linear):
+        """ Initialize the target polynomial critic network"""
+        # TODO: FIX TO REFLECT NEW CHANGES TO OBJECT.
```
Comment (philkuz) [inline experiments/exp1/polynomial_critic.py:103]: TODO here or no

File: experiments/exp1/polynomial_critic.py [hunk 4]
```diff
@@ -86,44 +79,52 @@ def setup_graph(self, state_input, action_input, net, linear):
...
 
-    def gradients(self,state_batch,action_batch):
-        """
-        Calculates the gradient of the batch
-        """
-        return self.sess.run(self.action_gradients,feed_dict={
-            self.state_input:state_batch,
-            self.action_input:action_batch
-            })[0]
+        self.loss = tf.reduce_mean(tf.square(diff)) #+ weight_decay
```
Comment (philkuz) [inline experiments/exp1/polynomial_critic.py:126]: You remove the gradients - why?

**Review by philkuz — COMMENTED** [review]
Mainly noticed a few loose todos and removing a sess.run from the gradients calculation. Just not sure why we did that.

File: experiments/exp1/polynomial_critic.py (hunk 2 above)
Comment (arkwave) [reply experiments/exp1/polynomial_critic.py:88]: pretty sure this todo is a leftover from when i was working on the polynomial critic. can be removed.
